core/trino-main/src/main/java/io/trino/sql/gen/JoinCompiler.java

```diff
@@ -1036,14 +1040,21 @@ public LookupSourceSupplier createLookupSourceSupplier(
                 LongArrayList addresses,
                 List<List<Block>> channels,
                 OptionalInt hashChannel,
+                List<Integer> joinChannels,
                 Optional<JoinFilterFunctionFactory> filterFunctionFactory,
                 Optional<Integer> sortChannel,
                 List<JoinFilterFunctionFactory> searchFunctionFactories,
                 HashArraySizeSupplier hashArraySizeSupplier)
         {
             PagesHashStrategy pagesHashStrategy = pagesHashStrategyFactory.createPagesHashStrategy(channels, hashChannel);
             try {
-                return constructor.newInstance(session, pagesHashStrategy, addresses, channels, filterFunctionFactory, sortChannel, searchFunctionFactories, hashArraySizeSupplier);
+                if (singleBigintColumn) {
```

Comment:
you can check `joinChannels` here directly

Reply:
^^